site/src/modules/resources/AgentExternal.tsx

```diff
@@ -0,0 +1,59 @@
+import { API } from "api/api";
+import { getErrorMessage } from "api/errors";
+import type { Workspace, WorkspaceAgent } from "api/typesGenerated";
+import isChromatic from "chromatic/isChromatic";
+import { CodeExample } from "components/CodeExample/CodeExample";
+import { displayError } from "components/GlobalSnackbar/utils";
+import { type FC, useEffect, useState } from "react";
+
+interface AgentExternalProps {
+	isExternalAgent: boolean;
+	agent: WorkspaceAgent;
+	workspace: Workspace;
+}
+
+export const AgentExternal: FC<AgentExternalProps> = ({
+	isExternalAgent,
+	agent,
+	workspace,
+}) => {
+	const [externalAgentToken, setExternalAgentToken] = useState<string | null>(
+		null,
+	);
+	const [command, setCommand] = useState<string | null>(null);
+
+	const origin = isChromatic() ? "https://example.com" : window.location.origin;
+	useEffect(() => {
+		if (
+			isExternalAgent &&
+			(agent.status === "timeout" || agent.status === "connecting")
+		) {
+			API.getWorkspaceAgentCredentials(workspace.id, agent.name)
+				.then((res) => {
+					setExternalAgentToken(res.agent_token);
+					setCommand(res.command);
+				})
+				.catch((err) => {
+					displayError(
+						getErrorMessage(err, "Failed to get external agent credentials"),
+					);
+				});
+		}
+	}, [isExternalAgent, agent.status, workspace.id, agent.name]);
```

Comment:
use `useQuery` for this. you can look at plenty of examples in the api/queries/ folder.